elements/lisk-p2p/test/unit/peer/inbound.ts

```diff
@@ -13,67 +13,140 @@
  *
  */
 import { expect } from 'chai';
-import { SCServerSocket } from 'socketcluster-server';
-import { InboundPeer } from '../../../src/peer';
+import { InboundPeer, PeerConfig } from '../../../src/peer';
 import { P2PDiscoveredPeerInfo } from '../../../src/p2p_types';
+import { SCServerSocket } from 'socketcluster-server';
+import { DEFAULT_RANDOM_SECRET } from '../../../src/constants';
 
 describe('peer/inbound', () => {
-	const DEFAULT_RANDOM_SECRET = 123;
-	const defaultPeerInfo: P2PDiscoveredPeerInfo = {
-		ipAddress: '12.12.12.12',
-		wsPort: 5001,
-		height: 545776,
-		isDiscoveredPeer: true,
-		version: '1.1.1',
-		protocolVersion: '1.1',
-	};
-
-	const socket = <SCServerSocket>({
-		on: sandbox.stub(),
-	} as any);
-
-	const defaultPeer = new InboundPeer(defaultPeerInfo, socket, {
-		rateCalculationInterval: 1000,
-		wsMaxMessageRate: 1000,
-		wsMaxMessageRatePenalty: 10,
-		secret: DEFAULT_RANDOM_SECRET,
-		maxPeerInfoSize: 10000,
-		maxPeerDiscoveryResponseLength: 1000,
+	let defaultPeerInfo: P2PDiscoveredPeerInfo;
+	let defaultPeerConfig: PeerConfig;
+	let defaultInboundPeer: InboundPeer;
+	let inboundSocket: SCServerSocket;
+
+	beforeEach(() => {
+		defaultPeerInfo = {
+			ipAddress: '12.12.12.12',
+			wsPort: 5001,
+			height: 545776,
+			isDiscoveredPeer: true,
+			version: '1.1.1',
+			protocolVersion: '1.1',
+		};
+		defaultPeerConfig = {
+			rateCalculationInterval: 1000,
+			wsMaxMessageRate: 1000,
+			wsMaxMessageRatePenalty: 10,
+			secret: DEFAULT_RANDOM_SECRET,
+			maxPeerInfoSize: 10000,
+			maxPeerDiscoveryResponseLength: 1000,
+		};
+		inboundSocket = <SCServerSocket>({
+			on: sandbox.stub(),
+			off: sandbox.stub(),
+			emit: sandbox.stub(),
+			destroy: sandbox.stub(),
+		} as any);
+		sandbox.spy(global, 'setTimeout');
```

Comment:
should we move it to `#constructor` test case like you did for `clearTimeout` in the below test case for consistency

Reply:
I cannot because I have to create the stub before we create the class.

Reply:
@shuse2 if we don't quickly find the reason why this is happening, I'll open a small independent issue to solve it properly.

Reply:
Addressed.